commons/src/main/java/com/powsybl/commons/json/JsonUtil.java

```diff
@@ -173,7 +197,7 @@ public static void writeOptionalIntegerField(JsonGenerator jsonGenerator, String
             for (Extension<T> extension : extendable.getExtensions()) {
                 ExtensionJsonSerializer serializer = supplier.findProvider(extension.getName());
                 if (serializer != null) {
-                    if (!headerDone) {
+                    if (!headerDone && headerWanted) {
```

Comment:
In the JSON converter,I chose to write extensions are at the end of the file to match the structure of xml and I would like to have something like this :

```javascript
extensions : [ {
       "id":" ID1",
       "extension1": { },
       "extension2": { }
}, {
      "id": "ID2",
      ...
} ]
```
To do so, it should be possible to write all the extensions for a given identifiable without the extensions header or it would gave something like this : 

```javascript
extensions : [ {
       "id": "ID1",
        extensions : {
             "extension1": { },
             "extension2": { }
        }
}, {
      "id": "ID2",
      ...
} ]
```